libr/debug/debug.c

```diff
@@ -343,6 +343,11 @@ R_API int r_debug_select(RDebug *dbg, int pid, int tid) {
 	if (dbg->h && dbg->h->select && !dbg->h->select (pid, tid))
 		return false;
 
+	{
+		char buf[32];
+		sprintf (buf, "pid %d", pid);
+		r_io_system (dbg->iob.io, buf);
```

Comment:
it will not leak memory because sdb_fmt have some static buffers that are suposed to be used in the same stackframe of the program. the '0' specifies which buffer you want to use. see shlr/sdb/src/fmt.c